src/core_plugins/i18n/server/__tests__/i18n.js

```diff
@@ -0,0 +1,227 @@
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!',
+          'test_plugin_2-XXXXXX': 'This is XXXXXX string',
+          'test_plugin_2-YYYY_PPPP': 'This is YYYY_PPPP string',
+          'test_plugin_2-FFFFFFFFFFFF': 'This is FFFFFFFFFFFF string',
+          'test_plugin_2-ZZZ': 'This is ZZZ string'
+        };
+        return checkTranslations(expectedTranslationJson, languageTag);
+      });
+
+      it('should return the translations for de locale as registered' , function () {
+        const languageTag = ['de'];
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        return checkTranslations(expectedTranslationJson, languageTag);
+      });
+
+      it('should return the most recently registered translation for a key that has multiple translations' , function () {
+        const pluginName = 'test_plugin_2';
+        const pluginTranslationPath = join(FIXTURES, 'translations', pluginName);
+        const translationFiles = [ join(pluginTranslationPath, 'de.json') ];
+        translationFiles.forEach(i18n.registerTranslations);
+        const languageTag = ['de'];
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS! I am regsitered afterwards!',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        return checkTranslations(expectedTranslationJson, languageTag);
+      });
+
+    });
+
+    describe('getRegisteredTranslationLocales', function () {
+
+      it('should return all locales registered', function () {
+        const expectedLocales = ['en', 'de'];
+        return checkRegisteredLocales(expectedLocales);
+      });
+
+    });
+
+  });
+
+  describe('registerTranslations', function () {
+
+    it('should throw error when regsitering empty filename', function () {
+      return expect(i18n.registerTranslations).withArgs('').to.throwError();
+    });
+
+    it('should throw error when regsitering filename with no extension', function () {
+      return expect(i18n.registerTranslations).withArgs('file1').to.throwError();
+    });
+
+    it('should throw error when regsitering filename with non JSON extension', function () {
+      return expect(i18n.registerTranslations).withArgs('file1.txt').to.throwError();
+    });
+
+  });
+
+});
+
+function checkTranslationsForDefaultLocale(expectedTranslations) {
+  return i18n.getTranslationsForDefaultLocale()
+  .then(function (actualTranslations) {
+    expect(_.isEqual(actualTranslations, expectedTranslations)).to.be(true);
+  });
+}
+
+function checkTranslations(expectedTranslations, languageTags) {
+  return i18n.getTranslations.apply(this, languageTags)
```

Comment:
That was my bad. I had an issue on the API when i defined the rest opeator. Fixed now.